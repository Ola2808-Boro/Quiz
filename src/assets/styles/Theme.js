export const theme = {
  colors: {
    backgroundColor: "#F5F7FB",
    text: "#293264",
    buttonColor: "#F5F7FB",
    circleBottom: "#DEEBF8",
    circleUp: "#FFFAD1",
    buttonBg: "#4D5B9E",
    borderAnswer: "#4D5B9E",
  },

  fontSize: {
    huge: "70px",
    xlgrand: "44px",
    lgrand: "40px",
    grand: "37px",
    xxxxl: "34px",
    xxxl: "33px",
    xxl: "30 px",
    xl: "26px",
    mediumxl: "24px",
    l: "22px",
    ml: "18px",
    m: "14px",
    s: "10px",
  },
};
