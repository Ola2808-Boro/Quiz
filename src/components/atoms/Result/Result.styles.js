import styled from "styled-components";
import { device } from "helpers/breakpoints";
export const StyledResultText = styled.p`
  font-size: ${({ theme }) => theme.fontSize.m};
  color: ${({ theme }) => theme.colors.text};
  letter-spacing: 2px;
  padding: 1rem;
  z-index: 1;
  @media ${device.mobileS} {
    font-size: ${({ theme }) => theme.fontSize.l};
  }

  @media ${device.mobileL} {
    font-size: ${({ theme }) => theme.fontSize.xl};
  }
  @media ${device.tablet} {
    font-size: ${({ theme }) => theme.fontSize.xxl};
  }
  @media ${device.laptop} {
    font-size: ${({ theme }) => theme.fontSize.xxxl};
  }

  @media ${device.desktop} {
    font-size: ${({ theme }) => theme.fontSize.xxxl};
  }
`;
