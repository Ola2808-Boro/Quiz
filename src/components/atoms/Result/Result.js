import React from "react";
import PropTypes from "prop-types";
import { StyledResultText } from "./Result.styles";
const Result = ({ result, resultMax }) => {
  return (
    <StyledResultText>
      You scored {result}/{resultMax.result} correct answers
    </StyledResultText>
  );
};

Result.propTypes = {
  result: PropTypes.string.isRequired,
  resultMax: PropTypes.number.isRequired,
};

export default Result;
