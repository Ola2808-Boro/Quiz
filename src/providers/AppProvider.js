import React from 'react';
import { GlobalStyle } from 'assets/styles/GlobalsStyle'
import { ThemeProvider } from 'styled-components';
import 'assets/styles/fonts.css'
import {theme} from 'assets/styles/Theme'


const AppProviders = ({ children }) => {
    return (
 
          <ThemeProvider theme={theme}>      
                <GlobalStyle />
                {children}
          </ThemeProvider>
 
    );
  };
  
  export default AppProviders;