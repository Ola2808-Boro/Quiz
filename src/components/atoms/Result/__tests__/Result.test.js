import { screen, render } from "@testing-library/react";
import "jest-styled-components";
import renderer from "react-test-renderer";
import Result from "components/atoms/Result/Result";
import { StyledResultText } from "components/atoms/Result/Result.styles";
import AppProvider from "providers/AppProvider";

describe("Test result info", () => {
  it("default style", () => {
    const tree = renderer
      .create(
        <AppProvider>
          <StyledResultText />
        </AppProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Test show result", async () => {
    render(
      <AppProvider>
        <Result result="5" resultMax={{ result: "10", isMaxResult: false }} />
      </AppProvider>
    );
    expect(screen.getByText(/5/i)).toBeInTheDocument();
  });
});
