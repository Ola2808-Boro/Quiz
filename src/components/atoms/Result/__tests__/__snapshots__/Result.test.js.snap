// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test result info default style 1`] = `
<p
  className="hcRIIt"
/>
`;
